### DISCUSSION TITLE
Auth0 Provider beforeLoad Implementation

### DISCUSSION THREAD

**Discussion by allan-vera** [TanStack/router#1322] [General]
Hello. I am running into an issue where my `beforeLoad` function is running before my AuthProvider finishes loading.

My Auth Provider hook provides the following values: `isLoading`, `isAuthenticated`, and the `user` object.

I pass in `isAuthenticated` to the Router via its context. However, by default `isAuthenticated` is `false` while it loads. By the time my `protectRoute` function runs (my `beforeLoad` function), its still loading and thus get redirected to /login. How do get it to wait until the AuthProvider has finished loading?

**Comment by SeanCassiere** [comment]
You could conditionally render the RouterProvider after the authentication library has finished loading.
I'm doing something similar with `react-oidc-context`.

```tsx
function App() {
  const auth = useAuth();

  if (auth.isLoading) {
    return <AuthLoadingMessage />
  }
  
  return <RouterProvider router={router} context={{ auth }} />
}
```

**Reply by allan-vera** [reply]
Yeah I ultimately ended up doing something like that. Thanks for your input!
